Fix missing plugins javadoc and maven plugins for driver module

porunov opened the pull request:
Fixes #1950

Signed-off-by: Oleksandr Porunov <alexandr.porunov@gmail.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[doc only]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

porunov commented:
@farodin91 @FlorianHockmann @pluradj this is a blocking issue for the `0.5.0` release. Reviews would be very helpful for this PR because we won't be able to make `0.5.0` release without fixing #1950

FlorianHockmann commented:
Could you give a quick explanation of why these dependencies had to be moved from janusgraph-core to janusgraph-driver? Was this simply forgotten when janusgraph-driver was created as its own module?

porunov commented:
> Could you give a quick explanation of why these dependencies had to be moved from janusgraph-core to janusgraph-driver? Was this simply forgotten when janusgraph-driver was created as its own module?

@FlorianHockmann Sure. Below I provide reasons for each step which I decided to make when was fixing the dependency problems. 

**General intro and moving `maven-javadoc-plugin` into driver module**

To generate docs we should use `maven-javadoc-plugin` which was previously in `janusgraph-core`. Because all other modules were depending on `janusgraph-core`, the plugin was populated in all modules. In the commit https://github.com/JanusGraph/janusgraph/commit/b96aa6d26f74d3dc8f7404212bddeefe2d0790b4 we extracted JanusGraph Gremlin driver requirements in `janusgraph-driver` which doesn't depend on any modules but `janusgraph-core` now depends on `janusgraph-driver` module.
When the `janusgraph-driver` module was created, we forgot to move plugins which should be populated to all modules from `janusgraph-core` into `janusgraph-driver` (because now we can assume that `janusgraph-driver` is the root module because `janusgraph-core` directly depends on `janusgraph-driver` and all other modules indirectly depend on `janusgraph-driver`).
So, as I said, we should just move all necessary modules from old `"main" module (i.e. janusgraph-core)` into a new `"main" module (i.e. janusgraph-driver)`.

**Removing `gremlin-core` from driver and excluding `org.sonatype.gossip` it driver**

While moving those plugins I found that the dependency `gremlin-core` already exists in `janusgraph-driver`, so we don't need to write it in `janusgraph-driver` (so I removed it). You may notice that it excludes `org.sonatype.gossip` (https://github.com/jdillon/gossip) in `janusgraph-core` but doesn't excludes in `janusgraph-driver`, so I excluded it in `janusgraph-driver` because we are using `logback` for logging instead of `gossip`. Because we need only a single slf4j provider, we exclude `gossip`.

**Moving `org.codehaus.gmavenplus` from core into driver**

`org.codehaus.gmavenplus` (https://github.com/groovy/GMavenPlus) is just a plugin that allows you to integrate Groovy into your Maven projects.
For some reason it was added as a `core` dependency (Maybe because it has `gremlin-groovy` dependency) but nevertheless groovy was available in all other dependant modules. To not break this contract I just moved it into `janusgraph-driver` module.
This isn't necessary, but I thought that we may need it everywhere, so I moved it to `driver` module. I can move it back if you think that we don't need to populate this dependency into `driver` module.

**Removing `janusgraph-release` profile from `janusgraph-core`**

You may notice that `janusgraph-release` already exists in `janusgraph-driver`. That is why I removed it from `core`. But the profile in `driver` module didn't have some `maven-shade-plugin` plugin in `build` so I added it into `driver` module. Also, `maven-gpg-plugin` had `sign-artifacts` exclusion, so I added this exclusion into `driver` module.
I think `janusgraph-release` profile should be the same for all modules because it is connected to populating new JanusGraph releases.
